app/src/main/java/space/taran/arknavigator/ui/fragments/SettingsFragment.kt

```diff
@@ -0,0 +1,196 @@
+import android.view.View
+import android.view.ViewGroup
+import android.widget.RadioButton
+import android.widget.RadioGroup
+import androidx.annotation.ColorRes
+import androidx.core.content.ContextCompat
+import androidx.core.view.children
+import moxy.MvpAppCompatFragment
+import moxy.ktx.moxyPresenter
+import space.taran.arknavigator.R
+import space.taran.arknavigator.databinding.FragmentSettingsBinding
+import space.taran.arknavigator.mvp.model.UserPreferences.*
+import space.taran.arknavigator.mvp.presenter.SettingsPresenter
+import space.taran.arknavigator.mvp.view.SettingsView
+import space.taran.arknavigator.ui.App
+import space.taran.arknavigator.ui.activity.MainActivity
+import space.taran.arknavigator.ui.fragments.utils.Notifications
+import space.taran.arknavigator.utils.SETTINGS_SCREEN
+import space.taran.arknavigator.utils.showInfoDialog
+import java.lang.AssertionError
+
+class SettingsFragment : MvpAppCompatFragment(), SettingsView {
+
+    private lateinit var binding: FragmentSettingsBinding
+
+    private val presenter by moxyPresenter {
+        SettingsPresenter().apply {
+            Log.d(SETTINGS_SCREEN, "creating SettingsPresenter")
+            App.instance.appComponent.inject(this)
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        super.onCreateView(inflater, container, savedInstanceState)
+
+        Log.d(SETTINGS_SCREEN, "inflating layout for SettingsFragment")
+        binding = FragmentSettingsBinding.inflate(inflater, container, false)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        Log.d(SETTINGS_SCREEN, "view created in ResourcesFragment")
+        super.onViewCreated(view, savedInstanceState)
+
+        App.instance.appComponent.inject(this)
+
+        binding.apply {
+            crashGroup.setOnCheckedChangeListener { rg, checkedID ->
+                presenter.onCrashPreferenceClick(crashReportFromButton(checkedID))
+                rgButtonSelected(rg, checkedID)
+            }
+
+            imgReplicationGroup.setOnCheckedChangeListener { rg, checkedID ->
+                presenter.onImgCacheReplicationClick(imgReplicationFromButton(checkedID))
+                rgButtonSelected(rg, checkedID)
+            }
+
+            indexReplicationGroup.setOnCheckedChangeListener { rg, checkedID ->
+                presenter.onIndexReplicationClick(indexReplicationFromButton(checkedID))
+                rgButtonSelected(rg, checkedID)
+            }
+
+            crashInfo.setOnClickListener {
+                showInfoDialog(
+                    requireContext(),
+                    title = R.string.what_are_crash_reports_,
+                    descrText = R.string.crash_reports_explanation
+                )
+            }
+
+            //todo: insert meaningful explanation for images cache replication
```

Comment:
We need to get rid of other `//todo`s instead of putting new ones. If something should be done, we must use issue tracker for it. Particularly, about these 2 `//todo`s — we can just remove them.